packages/core/strapi/lib/services/entity-service/types/params/filters/attributes.ts

```diff
@@ -0,0 +1,92 @@
+import type { Attribute, Common, Utils } from '@strapi/strapi';
+
+export type NonFilterableKind = Extract<Attribute.Kind, 'password' | 'dynamiczone'>;
+export type FilterableKind = Exclude<Attribute.Kind, NonFilterableKind>;
+
+export type GetNonFilterableKeys<TSchemaUID extends Common.UID.Schema> = Utils.Object.KeysBy<
+  Attribute.GetAll<TSchemaUID>,
+  Attribute.OfType<NonFilterableKind>
+>;
+
+export type GetScalarKeys<TSchemaUID extends Common.UID.Schema> = Exclude<
+  Attribute.GetKeysByType<TSchemaUID, Attribute.NonPopulatableKind>,
+  GetNonFilterableKeys<TSchemaUID>
+>;
+
+export type GetNestedKeys<TSchemaUID extends Common.UID.Schema> = Exclude<
+  Attribute.GetKeysWithTarget<TSchemaUID>,
+  GetNonFilterableKeys<TSchemaUID>
+>;
+
+export type BooleanValue = boolean | 'true' | 'false' | 't' | 'f' | '1' | '0' | 1 | 0;
+
+export type NumberValue = string | number;
+
+export type DateValue = globalThis.Date | string | number;
+
+export type TimeValue = globalThis.Date | string | number;
+
+export type DateTimeValue = globalThis.Date | string | number;
+
+export type TimeStampValue = globalThis.Date | string | number;
+
+/**
+ * List of possible values for the scalar attributes
+ * Uses the local GetValue to benefit from the values' overrides
+ */
+export type ScalarValues = GetValue<
+  | Attribute.BigInteger
+  | Attribute.Boolean
+  | Attribute.DateTime
+  | Attribute.Date
+  | Attribute.Decimal
+  | Attribute.Email
+  | Attribute.Enumeration<string[]>
+  | Attribute.Float
+  | Attribute.Integer
+  | Attribute.JSON
+  // /!\  Password attributes are NOT filterable and should NOT be part of this union type.
+  //      The member below has been commented on purpose to avoid adding it back without noticing.
+  // | Attribute.Password
+  | Attribute.RichText
+  | Attribute.String
+  | Attribute.Text
+  | Attribute.Time
+  | Attribute.Timestamp
+  | Attribute.UID<Common.UID.Schema | undefined>
+>;
+
+/**
+ * Attribute.GetValue override with filter values
+ *
+ * Fallback to unknown if never is found
+ */
+export type GetValue<TAttribute extends Attribute.Attribute> = Utils.Expression.If<
+  Utils.Expression.IsNotNever<TAttribute>,
+  Utils.Expression.MatchFirst<
+    [
+      // Boolean
+      [Utils.Expression.Extends<TAttribute, Attribute.Boolean>, BooleanValue],
+      // Number
+      [
+        Utils.Expression.Extends<
+          TAttribute,
+          Attribute.Integer | Attribute.BigInteger | Attribute.Float | Attribute.Decimal
```

Comment:
BigInteger would be a string I believe

Reply:
NumberValue is string | number, so that would work